src/app/view/perspectives/post-it/post-it-perspective.component.ts

```diff
@@ -17,465 +17,271 @@
-    if (selection.documentIdx + 1 < this.postIts.length) {
-      const selectedDocument = this.documentComponents.toArray()[selection.documentIdx + 1];
-      this.lastClickedPostIt = this.postIts[selection.documentIdx + 1];
-      selectedDocument.select(0, selection.row);
+  private loadMoreOnInfiniteScroll(): void {
+    if (!this.allLoaded && this.navigationManager && this.navigationManager.validNavigation()) {
+      this.getPostIts();
     }
   }
 
-  private tryToSelectDocumentOnUp(selection: AttributePropertySelection): void {
-    if (selection.documentIdx - this.documentsPerRow() >= 0) {
-      const selectedDocument = this.documentComponents.toArray()[selection.documentIdx - this.documentsPerRow()];
-      this.lastClickedPostIt = this.postIts[selection.documentIdx - this.documentsPerRow()];
-      selectedDocument.select(selection.column, Number.MAX_SAFE_INTEGER);
-    }
+  private reinitializePostIts(): void {
+    this.resetToInitialState();
+    this.getPostIts();
   }
 
-  private tryToSelectDocumentOnDown(selection: AttributePropertySelection): void {
-    if (selection.documentIdx + this.documentsPerRow() < this.postIts.length) {
-      const selectedDocument = this.documentComponents.toArray()[selection.documentIdx + this.documentsPerRow()];
-      this.lastClickedPostIt = this.postIts[selection.documentIdx + this.documentsPerRow()];
-      selectedDocument.select(selection.column, 0);
-    }
+  private resetToInitialState(): void {
+    this.allLoaded = false;
+    this.page = 0;
+    this.postIts.splice(0);
+    this.pageSubscriptions.forEach(subscription => subscription.unsubscribe());
   }
 
-  private documentsPerRow(): number {
-    const postItWidth = 225;
-    const layoutWidth = this.layoutElement.nativeElement.clientWidth;
+  private getPostIts(): void {
+    this.infiniteScrollManager.startLoading();
 
-    return Math.max(1, Math.floor(layoutWidth / postItWidth));
+    const queryModel = this.navigationManager.queryWithPagination(this.page++, this.editable);
+    this.fetchQueryDocuments(queryModel);
+    this.subscribeOnDocuments(queryModel);
   }
 
-  private queryPage(pageNumber: number): Query {
-    const addDocumentPresent = this.editable && pageNumber === 0 ? 1 : 0;
-
-    return {
-      pageSize: this.documentsPerRow() * 2 - addDocumentPresent,
-      page: pageNumber,
-      filters: this.query.filters,
-      fulltext: this.query.fulltext,
-      collectionCodes: this.query.collectionCodes
-    };
+  public fetchQueryDocuments(queryModel: QueryModel): void {
+    this.store.dispatch(new DocumentsAction.Get({query: queryModel}));
   }
 
-  private fetchPostIts(): void {
-    if (this.fetchingData || !this.query || !this.hasWorkspace() || this.allLoaded) {
-      return;
-    }
+  private subscribeOnDocuments(queryModel: QueryModel) {
+    const subscription = this.store.select(selectDocumentsByCustomQuery(queryModel)).pipe(
+      skipWhile(() => !this.navigationManager.validNavigation())
```

Comment:
Relying on some external state is not the best idea in the observable. We should work with the values from the observable when checking such conditions.